src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -61,19 +71,71 @@ CHLSLCompiler::~CHLSLCompiler()
 }
 
 
-struct DxcCompilationResult
+static void try_upgrade_hlsl_version(std::vector<std::wstring>& arguments) 
 {
-    ComPtr<IDxcBlobEncoding> errorMessages;
-    ComPtr<IDxcBlob> objectBlob;
-    ComPtr<IDxcResult> compileResult;
+    auto foundShaderStageArgument = std::find(arguments.begin(), arguments.end(), L"-HV");
+    if (foundShaderStageArgument != arguments.end()) {
+        auto foundShaderStageArgumentValueIdx = foundShaderStageArgument - arguments.begin() + 1;
+        std::wstring version = arguments[foundShaderStageArgumentValueIdx];
+        if (version.length() >= 4 && (version[2] < '2' || (version[2] == '2' && version[3] <= '0'))) 
```

Comment:
use `atoi` or some other string2int function then compare against being less than `2021`

(also check for last [back()] char being a letter instead of a number, then ignore the check)